sjsonnet/src/sjsonnet/PrettyYamlRenderer.scala

```diff
@@ -0,0 +1,434 @@
+package sjsonnet
+
+import java.io.{StringWriter, Writer}
+import java.util.regex.Pattern
+
+import ujson.BaseRenderer
+import upickle.core.{ArrVisitor, ObjVisitor}
+
+/**
+ * A version of YamlRenderer that tries its best to make the output YAML as
+ * pretty as possible: unquoted strings, de-dented lists, etc. Follows the PyYAML
+ * style. Also adds the ability to stream writes to a generic `Writer`.
+ */
+class PrettyYamlRenderer(out: Writer = new java.io.StringWriter(),
+                         indentArrayInObject: Boolean = false,
+                         indent: Int = 2,
+                         idealWidth: Int = 80) extends BaseRenderer[Writer](out, indent){
+  var newlineBuffered = false
+  var dashBuffered = false
+  var afterColon = false
```

Comment:
`YamlRenderer` uses backtracking to go back and delete any spaces that are followed by a newline. In order to make `PrettyYamlRenderer` work with opaque Writers that might not support such backtracking, I instead buffer up the space-after-colon in this `afterColon` boolean and only flush it to `out` when I know I am writing something that's not a newline following the colon